### PR TITLE
scalable arch selection based on KernelSel4Arch

### PR DESCRIPTION
**Pull request by axel-h**
Simplify the arch selection process by introducing a table that uses KernelSel4Arch. This variable hold all required information in one place, so it seem the best choice fr the current needs. Also, the build process will fail now here if the architecture is unsupported instead of silently ignoring this here (and then maybe fail later with a more obscure error). The table can be extended easily in the future for new new architectures.

### CODE REVIEW

**Comment by axel-h** [comment]
@ssrg-bamboo `test`

**Comment by ssrg-bamboo** [comment]
Hello, I'm a bot! I'll bring this PR into Trustworthy Systems and run some tests

**Comment by ssrg-bamboo** [comment]
## Test results for commit 4ad7f8dc6a1385eb1d85bf99ed0e854fdfa06464

### Summary

| Test name | Result |
| --- | --- |
| Code Quality - Check licenses - musllibc - EXT-axel-h-patch-axel-1-9 | SUCCESSFUL |
| Code Quality - Style - musllibc - EXT-axel-h-patch-axel-1-9 | FAILED |

### Detailed failure logs

| Project | Stage | Result summary |
| --- | --- | --- |
| Code Quality - Style - musllibc - EXT-axel-h-patch-axel-1-9 | Style check on changed files | 1 of 3 jobs failed |

**Job: style**
Build logs

    15-Feb-2021 05:48:03	Build Code Quality - Style - musllibc - EXT-axel-h-patch-axel-1-9 - style #1 (CQ-MUSLLIBCSTYLE16-MUSLLIBCSTYLEPR-1) started building on agent A local agent 2, bamboo version: 7.1.4
    15-Feb-2021 05:48:03	I run on the Bamboo server!
    15-Feb-2021 05:48:03	Build working directory is /var/atlassian/application-data/bamboo/xml-data/build-dir/151224373/CQ-MUSLLIBCSTYLE16-MUSLLIBCSTYLEPR
    15-Feb-2021 05:48:03	Executing build Code Quality - Style - musllibc - EXT-axel-h-patch-axel-1-9 - style #1 (CQ-MUSLLIBCSTYLE16-MUSLLIBCSTYLEPR-1)
    15-Feb-2021 05:48:03	Preparing artifact 'code' for use at /var/atlassian/application-data/bamboo/xml-data/build-dir/151224373/CQ-MUSLLIBCSTYLE16-MUSLLIBCSTYLEPR/. (location: .)
    15-Feb-2021 05:48:03	Artifact 'code' prepared in 10.70 ms at /var/atlassian/application-data/bamboo/xml-data/build-dir/151224373/CQ-MUSLLIBCSTYLE16-MUSLLIBCSTYLEPR/./code.tar.gz
    15-Feb-2021 05:48:03	Running pre-build action: VCS Version Collector
    15-Feb-2021 05:48:03	Starting task 'Untar code' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    15-Feb-2021 05:48:03	+ tar xf code.tar.gz --recursive-unlink
    15-Feb-2021 05:48:03	Finished task 'Untar code' with result: Success
    15-Feb-2021 05:48:03	Starting task 'Set permissions so Bamboo can cleanup' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    15-Feb-2021 05:48:03	+ chmod -R o+rwx /var/atlassian/application-data/bamboo/xml-data/build-dir/151224373/CQ-MUSLLIBCSTYLE16-MUSLLIBCSTYLEPR
    15-Feb-2021 05:48:03	Finished task 'Set permissions so Bamboo can cleanup' with result: Success
    15-Feb-2021 05:48:03	Starting task 'Check style of changed files' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    15-Feb-2021 05:48:03	+ cd musllibc
    15-Feb-2021 05:48:03	+ xargs -d '\n' /var/atlassian/application-data/bamboo/xml-data/build-dir/151224373/CQ-MUSLLIBCSTYLE16-MUSLLIBCSTYLEPR/standard_seL4_tools/misc/style.sh
    15-Feb-2021 05:48:03	++ git merge-base 4ad7f8dc6a1385eb1d85bf99ed0e854fdfa06464 4a8335b2248d934e2e40386af4f1b0495b3c173d
    15-Feb-2021 05:48:03	+ git diff --name-only 4ad7f8dc6a1385eb1d85bf99ed0e854fdfa06464 4a8335b2248d934e2e40386af4f1b0495b3c173d
    15-Feb-2021 05:48:03	+ git diff --exit-code
    15-Feb-2021 05:48:03	diff --git a/CMakeLists.txt b/CMakeLists.txt
    15-Feb-2021 05:48:03	index 2248dd88..f3dfc72b 100644
    15-Feb-2021 05:48:03	--- a/CMakeLists.txt
    15-Feb-2021 05:48:03	+++ b/CMakeLists.txt
    15-Feb-2021 05:48:03	@@ -17,14 +17,15 @@ project(libmuslc NONE)
    15-Feb-2021 05:48:03	 set(LIB_MUSLC_PARALLEL_BUILD true CACHE BOOL "Use `make -j` when building libmuslc")
    15-Feb-2021 05:48:03	 
    15-Feb-2021 05:48:03	 # Map KernelSel4Arch to muslc architecture.
    15-Feb-2021 05:48:03	-set(arch_mapping     # element format: "KernelSel4Arch[:muslc_arch]"
    15-Feb-2021 05:48:03	+set(
    15-Feb-2021 05:48:03	+    arch_mapping # element format: "KernelSel4Arch[:muslc_arch]"
    15-Feb-2021 05:48:03	     "ia32:i386"
    15-Feb-2021 05:48:03	-    "x86_64"         # use KernelSel4Arch
    15-Feb-2021 05:48:03	+    "x86_64" # use KernelSel4Arch
    15-Feb-2021 05:48:03	     "aarch32:arm"
    15-Feb-2021 05:48:03	-    "arm_hyp:arm"    # arm_hyp is legacy hack that might get removed one day
    15-Feb-2021 05:48:03	-    "aarch64"        # use KernelSel4Arch
    15-Feb-2021 05:48:03	-    "riscv32:riscv"  # no difference between 32/64 bit
    15-Feb-2021 05:48:03	-    "riscv64:riscv"  # no difference between 32/64 bit
    15-Feb-2021 05:48:03	+    "arm_hyp:arm" # arm_hyp is legacy hack that might get removed one day
    15-Feb-2021 05:48:03	+    "aarch64" # use KernelSel4Arch
    15-Feb-2021 05:48:03	+    "riscv32:riscv" # no difference between 32/64 bit
    15-Feb-2021 05:48:03	+    "riscv64:riscv" # no difference between 32/64 bit
    15-Feb-2021 05:48:03	 )
    15-Feb-2021 05:48:03	 # Find the tuple with the mapping information. If KernelSel4Arch is not set at
    15-Feb-2021 05:48:03	 # all, matching will fail.
    15-Feb-2021 05:48:03	@@ -37,8 +38,8 @@ else()
    15-Feb-2021 05:48:03	 endif()
    15-Feb-2021 05:48:03	 
    15-Feb-2021 05:48:03	 message(
    15-Feb-2021 05:48:03	-    STATUS
    15-Feb-2021 05:48:03	-    "${PROJECT_NAME} architecture: '${muslc_arch}' (from KernelSel4Arch '${KernelSel4Arch}')")
    15-Feb-2021 05:48:03	+    STATUS "${PROJECT_NAME} architecture: '${muslc_arch}' (from KernelSel4Arch '${KernelSel4Arch}')"
    15-Feb-2021 05:48:03	+)
    15-Feb-2021 05:48:03	 
    15-Feb-2021 05:48:03	 set(
    15-Feb-2021 05:48:03	     LIB_MUSLC_GLOB_SOURCES OFF
    15-Feb-2021 05:48:03	@@ -83,45 +84,67 @@ get_property(compile_options DIRECTORY PROPERTY COMPILE_OPTIONS)
    15-Feb-2021 05:48:03	 # CMAKE_C_FLAGS_DEBUG, CMAKE_C_FLAGS_RELEASE, CMAKE_C_FLAGS_RELWITHDEBINFO, CMAKE_C_FLAGS_MINSIZEREL
    15-Feb-2021 05:48:03	 string(TOUPPER CMAKE_BUILD_TYPE build_type)
    15-Feb-2021 05:48:03	 set(cmake_built_type_config_name CMAKE_C_FLAGS_${build_type})
    15-Feb-2021 05:48:03	-separate_arguments(cmake_c_flags_sep NATIVE_COMMAND "${CMAKE_C_FLAGS};${${cmake_built_type_config_name}}")
    15-Feb-2021 05:48:03	+separate_arguments(
    15-Feb-2021 05:48:03	+    cmake_c_flags_sep NATIVE_COMMAND "${CMAKE_C_FLAGS};${${cmake_built_type_config_name}}"
    15-Feb-2021 05:48:03	+)
    15-Feb-2021 05:48:03	 
    15-Feb-2021 05:48:03	 list(APPEND compile_options "${cmake_c_flags_sep}")
    15-Feb-2021 05:48:03	 
    15-Feb-2021 05:48:03	 # Add the target triple to the compile flags if we are using clang
    15-Feb-2021 05:48:03	 # It is not part of the CMAKE_C_FLAGS variable
    15-Feb-2021 05:48:03	-if (CMAKE_C_COMPILER_ID STREQUAL "Clang")
    15-Feb-2021 05:48:03	+if(CMAKE_C_COMPILER_ID STREQUAL "Clang")
    15-Feb-2021 05:48:03	     list(APPEND compile_options "${CMAKE_C_COMPILE_OPTIONS_TARGET}${CMAKE_C_COMPILER_TARGET}")
    15-Feb-2021 05:48:03	 endif()
    15-Feb-2021 05:48:03	 
    15-Feb-2021 05:48:03	 include(memoize)
    15-Feb-2021 05:48:03	 # memoize this installation rule which will save the resulting artifact in a cache and reuse it across builds.
    15-Feb-2021 05:48:03	 # This will rebuild from source if the git directory has changes or has a changed commit hash.
    15-Feb-2021 05:48:03	-memoize_add_custom_command(musllibc "${CMAKE_CURRENT_BINARY_DIR}/build-temp/stage" "${CMAKE_CURRENT_SOURCE_DIR}" "${compile_options} ${muslc_arch}" ""
    15-Feb-2021 05:48:03	+memoize_add_custom_command(
    15-Feb-2021 05:48:03	+    musllibc
    15-Feb-2021 05:48:03	+    "${CMAKE_CURRENT_BINARY_DIR}/build-temp/stage"
    15-Feb-2021 05:48:03	+    "${CMAKE_CURRENT_SOURCE_DIR}"
    15-Feb-2021 05:48:03	+    "${compile_options} ${muslc_arch}"
    15-Feb-2021 05:48:03	+    ""
    15-Feb-2021 05:48:03	     OUTPUT
    15-Feb-2021 05:48:03	-        build-temp/stage/lib/libc.a
    15-Feb-2021 05:48:03	-        # If we have to rebuild, first clear the temporary build directory as
    15-Feb-2021 05:48:03	-        # we have no correctly captured the output files or dependencies
    15-Feb-2021 05:48:03	-    COMMAND rm -r build-temp
    15-Feb-2021 05:48:03	-    COMMAND mkdir -p build-temp
    15-Feb-2021 05:48:03	+    build-temp/stage/lib/libc.a
    15-Feb-2021 05:48:03	+    # If we have to rebuild, first clear the temporary build directory as
    15-Feb-2021 05:48:03	+    # we have no correctly captured the output files or dependencies
    15-Feb-2021 05:48:03	     COMMAND
    15-Feb-2021 05:48:03	-        ${CMAKE_COMMAND} -E env TOOLPREFIX=${CROSS_COMPILER_PREFIX}
    15-Feb-2021 05:48:03	-        # TODO: should also pass in the cflags for the CMAKE_BUILD_TYPE ?
    15-Feb-2021 05:48:03	-        # Suppress make output by default. Verbose make output can now be achieved
    15-Feb-2021 05:48:03	-        # by MAKEFLAGS=V3 ninja (${MAKEFLAGS:--s} expands to -s if MAKEFLAGS is unset).
    15-Feb-2021 05:48:03	-        C_COMPILER=${CMAKE_C_COMPILER}
    15-Feb-2021 05:48:03	-        "MAKEFLAGS=\$\${MAKEFLAGS:--s}"
    15-Feb-2021 05:48:03	-        NK_CFLAGS="${compile_options}" TARGET=${muslc_arch} make -C build-temp ${parallel} -f
    15-Feb-2021 05:48:03	-        "${CMAKE_CURRENT_SOURCE_DIR}/Makefile"
    15-Feb-2021 05:48:03	-        "STAGE_DIR=${CMAKE_CURRENT_BINARY_DIR}/build-temp/stage"
    15-Feb-2021 05:48:03	-        "SOURCE_DIR=${CMAKE_CURRENT_SOURCE_DIR}"
    15-Feb-2021 05:48:03	-    DEPENDS ${deps} COMMAND_EXPAND_LISTS
    15-Feb-2021 05:48:03	-    COMMENT "Invoking muslc build system"
    15-Feb-2021 05:48:03	+    rm
    15-Feb-2021 05:48:03	+    -r
    15-Feb-2021 05:48:03	+    build-temp
    15-Feb-2021 05:48:03	+    COMMAND
    15-Feb-2021 05:48:03	+    mkdir
    15-Feb-2021 05:48:03	+    -p
    15-Feb-2021 05:48:03	+    build-temp
    15-Feb-2021 05:48:03	+    COMMAND
    15-Feb-2021 05:48:03	+    ${CMAKE_COMMAND}
    15-Feb-2021 05:48:03	+    -E
    15-Feb-2021 05:48:03	+    env
    15-Feb-2021 05:48:03	+    TOOLPREFIX=${CROSS_COMPILER_PREFIX}
    15-Feb-2021 05:48:03	+    # TODO: should also pass in the cflags for the CMAKE_BUILD_TYPE ?
    15-Feb-2021 05:48:03	+    # Suppress make output by default. Verbose make output can now be achieved
    15-Feb-2021 05:48:03	+    # by MAKEFLAGS=V3 ninja (${MAKEFLAGS:--s} expands to -s if MAKEFLAGS is unset).
    15-Feb-2021 05:48:03	+    C_COMPILER=${CMAKE_C_COMPILER}
    15-Feb-2021 05:48:03	+    "MAKEFLAGS=\$\${MAKEFLAGS:--s}"
    15-Feb-2021 05:48:03	+    NK_CFLAGS="${compile_options}"
    15-Feb-2021 05:48:03	+    TARGET=${muslc_arch}
    15-Feb-2021 05:48:03	+    make
    15-Feb-2021 05:48:03	+    -C
    15-Feb-2021 05:48:03	+    build-temp
    15-Feb-2021 05:48:03	+    ${parallel}
    15-Feb-2021 05:48:03	+    -f
    15-Feb-2021 05:48:03	+    "${CMAKE_CURRENT_SOURCE_DIR}/Makefile"
    15-Feb-2021 05:48:03	+    "STAGE_DIR=${CMAKE_CURRENT_BINARY_DIR}/build-temp/stage"
    15-Feb-2021 05:48:03	+    "SOURCE_DIR=${CMAKE_CURRENT_SOURCE_DIR}"
    15-Feb-2021 05:48:03	+    DEPENDS
    15-Feb-2021 05:48:03	+    ${deps}
    15-Feb-2021 05:48:03	+    COMMAND_EXPAND_LISTS
    15-Feb-2021 05:48:03	+    COMMENT
    15-Feb-2021 05:48:03	+    "Invoking muslc build system"
    15-Feb-2021 05:48:03	 )
    15-Feb-2021 05:48:03	 
    15-Feb-2021 05:48:03	-add_custom_target(
    15-Feb-2021 05:48:03	-    muslc_gen
    15-Feb-2021 05:48:03	-    DEPENDS build-temp/stage/lib/libc.a
    15-Feb-2021 05:48:03	-)
    15-Feb-2021 05:48:03	+add_custom_target(muslc_gen DEPENDS build-temp/stage/lib/libc.a)
    15-Feb-2021 05:48:03	 
    15-Feb-2021 05:48:03	 add_library(muslc_imported STATIC IMPORTED GLOBAL)
    15-Feb-2021 05:48:03	 add_dependencies(muslc_imported muslc_gen)
    15-Feb-2021 05:48:03	Failing task since return code of [/var/atlassian/application-data/bamboo/temp/CQ-MUSLLIBCSTYLE16-MUSLLIBCSTYLEPR-1-ScriptBuildTask-2448051102791613392.sh] was 1 while expected 0
    15-Feb-2021 05:48:03	Finished task 'Check style of changed files' with result: Failed
    15-Feb-2021 05:48:03	Starting task 'Set permissions so Bamboo can cleanup' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    15-Feb-2021 05:48:03	+ chmod -R o+rwx /var/atlassian/application-data/bamboo/xml-data/build-dir/151224373/CQ-MUSLLIBCSTYLE16-MUSLLIBCSTYLEPR
    15-Feb-2021 05:48:03	Finished task 'Set permissions so Bamboo can cleanup' with result: Success
    15-Feb-2021 05:48:03	Starting task 'Run astyle code insights' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    15-Feb-2021 05:48:03	+ cd musllibc
    15-Feb-2021 05:48:03	++ cut -d/ -f4
    15-Feb-2021 05:48:03	++ echo ssh://git@bitbucket.keg.ertos.in.nicta.com.au:7999/sel4/musllibc.git
    15-Feb-2021 05:48:03	+ bb_proj=sel4
    15-Feb-2021 05:48:03	+ /scripts/bb_code_insights/create_code_insights.sh astyle sel4 musllibc
    15-Feb-2021 05:48:03	++ dirname /scripts/bb_code_insights/create_code_insights.sh
    15-Feb-2021 05:48:03	+ SCRIPT_DIR=/scripts/bb_code_insights
    15-Feb-2021 05:48:03	+ BBS_URL=https://bitbucket.ts.data61.csiro.au
    15-Feb-2021 05:48:03	+ BBS_PROJECT=sel4
    15-Feb-2021 05:48:03	+ BBS_REPO=musllibc
    15-Feb-2021 05:48:03	+ REPORT_TO_RUN=astyle
    15-Feb-2021 05:48:03	++ git rev-parse HEAD
    15-Feb-2021 05:48:03	+ COMMIT_ID=4ad7f8dc6a1385eb1d85bf99ed0e854fdfa06464
    15-Feb-2021 05:48:03	+ REPORT_KEY=astyle.report
    15-Feb-2021 05:48:03	+ set +x
    15-Feb-2021 05:48:03	+ report_file=astyle_report.json
    15-Feb-2021 05:48:03	+ annotations_file=astyle_annotations.json
    15-Feb-2021 05:48:03	+ rm -f astyle_report.json astyle_annotations.json
    15-Feb-2021 05:48:03	+ /scripts/bb_code_insights/astyle
    15-Feb-2021 05:48:03	Reporting on these files:  ['CMakeLists.txt']
    15-Feb-2021 05:48:03	+ echo 'Reporting the results to BitBucket, but doing so *secretly*, so people don'\''t see the access token'
    15-Feb-2021 05:48:03	Reporting the results to BitBucket, but doing so *secretly*, so people don't see the access token
    15-Feb-2021 05:48:03	+ set +x
    15-Feb-2021 05:48:04	+ '[' -f astyle_report.json ']'
    15-Feb-2021 05:48:04	+ cat astyle_report.json
    15-Feb-2021 05:48:04	+ '[' -f astyle_annotations.json ']'
    15-Feb-2021 05:48:04	+ cat astyle_annotations.json
    15-Feb-2021 05:48:04	{"data":[{"title":"Total number of files with incorrect style","value":1},{"title":"Total number of lines with incorrect style","value":32}],"createdDate":1613328483993,"key":"astyle.report","title":"Code style"}{"title": "Code style", "data": [{"title": "Total number of files with incorrect style", "value": 1}, {"title": "Total number of lines with incorrect style", "value": 32}]}{"annotations": [{"path": "CMakeLists.txt", "line": 20, "message": "There are 1 line(s) of incorrect style here.", "severity": "HIGH"}, {"path": "CMakeLists.txt", "line": 22, "message": "There are 1 line(s) of incorrect style here.", "severity": "HIGH"}, {"path": "CMakeLists.txt", "line": 24, "message": "There are 4 line(s) of incorrect style here.", "severity": "HIGH"}, {"path": "CMakeLists.txt", "line": 40, "message": "There are 2 line(s) of incorrect style here.", "severity": "HIGH"}, {"path": "CMakeLists.txt", "line": 86, "message": "There are 1 line(s) of incorrect style here.", "severity": "HIGH"}, {"path": "CMakeLists.txt", "line": 92, "message": "There are 1 line(s) of incorrect style here.", "severity": "HIGH"}, {"path": "CMakeLists.txt", "line": 99, "message": "There are 1 line(s) of incorrect style here.", "severity": "HIGH"}, {"path": "CMakeLists.txt", "line": 101, "message": "There are 5 line(s) of incorrect style here.", "severity": "HIGH"}, {"path": "CMakeLists.txt", "line": 107, "message": "There are 12 line(s) of incorrect style here.", "severity": "HIGH"}, {"path": "CMakeLists.txt", "line": 121, "message": "There are 4 line(s) of incorrect style here.", "severity": "HIGH"}]}
    15-Feb-2021 05:48:04	Finished task 'Run astyle code insights' with result: Success
    15-Feb-2021 05:48:04	Running post build plugin 'NCover Results Collector'
    15-Feb-2021 05:48:04	Running post build plugin 'Artifact Copier'
    15-Feb-2021 05:48:04	Running post build plugin 'npm Cache Cleanup'
    15-Feb-2021 05:48:04	Running post build plugin 'Clover Results Collector'
    15-Feb-2021 05:48:04	Running post build plugin 'Docker Container Cleanup'
    15-Feb-2021 05:48:04	Successfully removed working directory at '/var/atlassian/application-data/bamboo/xml-data/build-dir/151224373/CQ-MUSLLIBCSTYLE16-MUSLLIBCSTYLEPR'
    15-Feb-2021 05:48:04	Finalising the build...
    15-Feb-2021 05:48:04	Stopping timer.
    15-Feb-2021 05:48:04	Build CQ-MUSLLIBCSTYLE16-MUSLLIBCSTYLEPR-1 completed.
    15-Feb-2021 05:48:04	Running on server: post build plugin 'NCover Results Collector'
    15-Feb-2021 05:48:04	Running on server: post build plugin 'Build Hanging Detection Configuration'
    15-Feb-2021 05:48:04	Running on server: post build plugin 'Clover Delta Calculator'
    15-Feb-2021 05:48:04	Running on server: post build plugin 'Maven Dependencies Postprocessor'
    15-Feb-2021 05:48:04	All post build plugins have finished
    15-Feb-2021 05:48:04	Generating build results summary...
    15-Feb-2021 05:48:04	Saving build results to disk...
    15-Feb-2021 05:48:04	Store variable context...
    15-Feb-2021 05:48:04	Indexing build results...
    15-Feb-2021 05:48:04	Finished building CQ-MUSLLIBCSTYLE16-MUSLLIBCSTYLEPR-1.


This is the most I can report on right now, sorry!

**Comment by axel-h** [comment]
@ssrg-bamboo `test`

**Comment by ssrg-bamboo** [comment]
Hello, I'm a bot! I'll bring this PR into Trustworthy Systems and run some tests

**Comment by ssrg-bamboo** [comment]
All the tests we ran have passed! Nice job!

**Review by oliver-wm — APPROVED** [review]

File: Makefile.muslc [hunk 1]
```diff
@@ -93,8 +93,8 @@ $(ALL_LIBS) $(ALL_TOOLS) $(ALL_OBJS) $(ALL_OBJS:%.o=%.lo) $(GENH) $(GENH_INT): |
 $(OBJ_DIRS):
 	mkdir -p $@
 
-obj/include/bits/alltypes.h: $(srcdir)/arch/$(ARCH)/bits/alltypes.h.in $(srcdir)/include/alltypes.h.in $(srcdir)/tools/mkalltypes.sed
-	sed -f $(srcdir)/tools/mkalltypes.sed $(srcdir)/arch/$(ARCH)/bits/alltypes.h.in $(srcdir)/include/alltypes.h.in > $@
+obj/include/bits/alltypes.h: $(srcdir)/tools/mkalltypes.sed $(srcdir)/arch/$(ARCH)/bits/alltypes.h.in $(srcdir)/include/alltypes.h.in
+	$(srcdir)/tools/mkalltypes $@ $^
```
[the remark this answers is not in the document]
Comment (axel-h) [reply Makefile.muslc:97]: - $@ is the target obj/include/bits/alltypes.h
- $^ are all prereqisites ($(srcdir)/tools/mkalltypes.sed $(srcdir)/arch/$(ARCH)/bits/alltypes.h.in $(srcdir)/include/alltypes.h.in)
Params for mkalltypes are
- output file
- sed file
- templates to apply the sewd file on

**Comment by axel-h** [comment]
Do you want me to add more comment so this can be merged?

**Comment by oliver-wm** [comment]
All good!

**Pull request merged**